### PR TITLE
[DEV-3171] component-boilerplate isMounted bug and userIdent bug fixes

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> IsMounted check was causing data to not be returned due to race condition / mysterious dismount.
> Code is commented out for easy access to run tests, we can probably also delete it though.
> DS: Lets delete, git version history exists if we need to see the old code
## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3171
 - Process.st launch checklist: https://app.process.st/runs/DEV-3171-Component-Boilerplate-and-mint-update-o3odNUy7pswnNgLZQ4RPtA

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/component-boilerplate/src/hooks/graphql/useBaseQuery.ts [hunk 1]
```diff
@@ -85,12 +85,11 @@ export function useBaseQuery<T = any>(
   initialState: BaseQueryData<T>
 ): [BaseQueryData<T>, (variables: unknown) => unknown] {
   const client: GraphQLClient = useGraphQLClient();
-  const isMountedRef = useIsMountedRef();
+  // const isMountedRef = useIsMountedRef();
```
Comment (DerekSiemens) [inline packages/component-boilerplate/src/hooks/graphql/useBaseQuery.ts:88]: Flagging commented out code in this file for deletion

**Comment by DerekSiemens** [comment]
Needs changelog update!

File: packages/component-boilerplate/src/hooks/graphql/useBaseQuery.ts (hunk 1 above)
Comment (00salmon) [reply packages/component-boilerplate/src/hooks/graphql/useBaseQuery.ts]: deleted!

File: packages/component-boilerplate/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.2.3] - 2021-12-07
```
Comment (DerekSiemens) [inline packages/component-boilerplate/CHANGELOG.md:8]: Probably should make it 1.3.0

Comment (DerekSiemens) [inline packages/component-boilerplate/CHANGELOG.md:8]:
```suggestion
## [1.3.0] - 2021-12-09
```

File: packages/component-boilerplate/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/component-boilerplate",
-  "version": "1.2.2",
+  "version": "1.2.3-11",
```
Comment (DerekSiemens) [inline packages/component-boilerplate/package.json:3]:
```suggestion
  "version": "1.3.0",
```

File: packages/component-boilerplate/CHANGELOG.md [hunk 4]
```diff
@@ -90,8 +101,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - useShareEvent
       useForm
 
-[unreleased]: https://github.com/saasquatch/program-tools/compare/component-boilerplate@1.2.2...HEAD
-[1.2.1]: https://github.com/saasquatch/program-tools/compare/component-boilerplate@1.2.1...component-boilerplate@1.2.2
+[unreleased]: https://github.com/saasquatch/program-tools/compare/component-boilerplate@1.2.3...HEAD
```
Comment (DerekSiemens) [inline packages/component-boilerplate/CHANGELOG.md:104]:
```suggestion
[unreleased]: https://github.com/saasquatch/program-tools/compare/component-boilerplate@1.3.0...HEAD
```

File: packages/component-boilerplate/CHANGELOG.md [hunk 5]
```diff
@@ -90,8 +101,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+[1.2.3]: https://github.com/saasquatch/program-tools/compare/component-boilerplate@1.2.2...component-boilerplate@1.2.3
```
Comment (DerekSiemens) [inline packages/component-boilerplate/CHANGELOG.md:105]:
```suggestion
[1.3.0]: https://github.com/saasquatch/program-tools/compare/component-boilerplate@1.2.2...component-boilerplate@1.3.0
```

**Comment by DerekSiemens** [comment]
Looks good from a hygiene perspective, just waiting on Logans review

File: packages/component-boilerplate/src/environment/UserIdentityContext.ts [hunk 6]
```diff
@@ -187,7 +190,7 @@ export function useUserIdentity(): UserIdentity | undefined {
     | undefined;
 
   const validIdentity = userIdentityFromJwt(identity?.jwt);
-  if (!validIdentity) {
+  if (identity && !validIdentity) {
     // Likely that the JWT has expired
     setUserIdentity(undefined);
```
Comment (loganvolkers) [inline packages/component-boilerplate/src/environment/UserIdentityContext.ts:195]: I still like to believe that we don't need this cleanup step, but I'm fine with leaving it in.

```suggestion
    setUserIdentity(undefined);
```

**Review by loganvolkers — APPROVED** [review]
Looks good to me

**Pull request merged**